src/libraries/System.Private.CoreLib/src/System/Threading/Thread.Unix.cs

```diff
@@ -10,6 +10,18 @@ namespace System.Threading
 {
     public sealed partial class Thread
     {
+#if TARGET_WASI
+        internal static System.Threading.Tasks.Task RegisterPollable(int handle)
```

Comment:
@dicej here I renamed the methods to be more descriptive. 

`RegisterPollable` and `DispatchWasiEventLoop`

Reply:
I'm open to better proposals